compiler/global/scoping.ml

```diff
@@ -203,20 +203,20 @@ let rec types env ty =
     | Etypefunrefinement(k, n_opt, ty_arg, ty_res, e) -> 
         let ty_arg = types env ty_arg in
         let env =
-    match n_opt with
-    | None -> env
-    | Some(n) -> Rename.append (Rename.make (S.singleton n)) env in
-        let ty_res = types env ty_res in
-        Zelus.Etypefunrefinement(kind k, None, ty_arg, ty_res, expression_types env e)
-     | Etypevec(ty_arg, si) -> Zelus.Etypevec(types env ty_arg, size env si)
-    | Etypefun(k, n_opt, ty_arg, ty_res) ->
-       let ty_arg = types env ty_arg in
-       let env =
-	 match n_opt with
-	 | None -> env
-	 | Some(n) -> Rename.append (Rename.make (S.singleton n)) env in
-       let ty_res = types env ty_res in
-       Zelus.Etypefun(kind k, None, ty_arg, ty_res)
+          match n_opt with
```

Comment:
Please make sure unit tests #7 and #8 compile correctly. Translate.ml throws an error when trying to compile the unit tests.